tracing/tracing.go

```diff
@@ -0,0 +1,121 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// propagator to use
+var propagator propagation.TextMapPropagator = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}) // traceparent header
+type Option interface {
+	apply(*options)
+}
+type options struct {
+	address string
+}
+
+// WithAddress takes an address in the form of Host:Port
+func WithAddress(addr string) Option {
+	return addrOption{address: addr}
+}
+
+type addrOption struct {
+	address string
+}
+
+func (o addrOption) apply(opts *options) {
+	opts.address = o.address
+}
+
+// OtlpGrpcExporter uses the otlptracegrpc modules and the otlptrace module to produce a new exporter at our default addr
+func OtlpGrpcExporter(ctx context.Context, opts ...Option) sdktrace.SpanExporter {
+
+	DefaultCollectorHost := "localhost"
+	var defaultCollectorPort uint16 = 4317
+	addr := fmt.Sprintf("%s:%d", DefaultCollectorHost, defaultCollectorPort)
+
+	options := options{
+		address: addr,
+	}
+
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+	otlpClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithEndpoint(addr), //Not strictly needed as we use the defaults
+		otlptracegrpc.WithReconnectionPeriod(15*time.Second),
+		otlptracegrpc.WithInsecure(),
+	)
+
+	spanExporter, err := otlptrace.New(ctx, otlpClient)
+	if err != nil {
+		log.Fatal(err)
+		//Is doing a fatal error here too risky? No easy way to bubble up errors from here to the app using this.
+		//without making each of the WithXOption() takes an error as an arg as well.
+		return nil
+	}
+	return spanExporter
+
+}
+
+func JaegerExporter() (spanExporter sdktrace.SpanExporter) {
+	fmt.Println("Creating Jaeger Exporter")
```

Comment:
This should probably be done by the caller if needed.

Reply:
Honestly, all of the jaeger stuff can be removed. We don't use it in prod/dev it just makes local development easier as I don't have to create an opentelemetry sidecar and can just launch the jaeger-all-in-one to collect/view traces. 

I'll remove and then add back later as a separate module that can be imported if desired.